datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -579,11 +578,20 @@ impl StreamedBatch {
     }
 }
 
+/// The data stored in this BufferedBatch
+#[derive(Debug)]
+enum BufferedBatchData {
+    /// The batch is in memory
+    InMemory(RecordBatch),
```

Comment:
it might make sense to put `size_estimation` in this enum too to ensure it is only used with `InMemory`